src/Homely.AspNetCore.Mvc.Helpers/Extensions/IMvcBuilderExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Homely.AspNetCore.Mvc.Helpers.Controllers;
+using Homely.AspNetCore.Mvc.Helpers.Models;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Homely.AspNetCore.Mvc.Helpers.Extensions
+{
+    public static class IMvcBuilderExtensions
+    {
+        /// <summary>
+        /// Registers a common webapi home controller.
+        /// </summary>
+        /// <param name="builder">An interface for configuring essential MVC services.</param>
+        /// <param name="services">Specifies the contract for a collection of service descriptors.</param>
+        /// <param name="callingType">Type of the calling assembly. This would usually be the main assembly of the web site.</param>
+        /// <param name="asciiBanner">Optional: some text to display in the home controller output.</param>
+        /// <returns>Chaining: the interface for configuring essential MVC services.</returns>
+        public static IMvcBuilder AddAHomeController(this IMvcBuilder builder,
+                                                     IServiceCollection services,
+                                                     string asciiBanner = null,
+                                                     Assembly callingAssembly = null)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (callingAssembly == null)
+            {
+                callingAssembly = Assembly.GetCallingAssembly();
+            }
+
+            var banner = new HomeControllerBanner(asciiBanner, callingAssembly);
+            services.AddSingleton<IHomeControllerBanner>(banner);
+
+            builder.AddApplicationPart(typeof(HomeController).Assembly);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets up some common, default JsonSerializerOptions settings:<br/>
+        /// - CamelCase property names.
+        /// - Indented formatting.
+        /// - Ignore null properties which have values.
+        /// - Enums are rendered as string's ... not their backing number-value.
+        /// </summary>
+        /// <param name="builder"Mvc builder to help create the Mvc settings.></param>
+        /// <returns>IMvcBuilder: the builder, so we can more builder methods.</returns>
+        public static IMvcBuilder AddDefaultJsonOptions(this IMvcBuilder builder)
+        {
+            return builder.AddJsonOptions(options =>
+            {
+                options.JsonSerializerOptions.IgnoreNullValues = true;
+                options.JsonSerializerOptions.WriteIndented = true;
+                options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+                options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+            });
+        }

```

Comment:
date format?

Reply:
Not needed - the did it right.

REF: [S.T.Json announcement](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-apis/).
also [the MSDN docs ](https://docs.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support)talk about this _explicitly_.

> The System.Text.Json library parses and writes DateTime and DateTimeOffset values according to the ISO 8601:-2019 extended profile.

Both links have json output examples with Dates to show the format it's in.

Reply:
cool! but, as i said previously.. (not sure if you saw), are we defaulting to System.Text.Json now? see the JIRA card, i thought we agreed to stick with JSON.NET first, _then_ do another change where we cutover to System.Text.Json

Reply:
oh yeah - cause i did this work BEFORE the jira card (private time).

and didn't want to strip that out cause it was working.

if you want, i can strip it out.

Reply:
@PureKrome i'd prefer what you have here, long term.. but, it's not what we agreed on in card, so was checking.

i guess my main concern is because we don't have contract tests (yet), we won't know if we break stuff for clients..

what are your thoughts?

Reply:
As discusses in slack-chat, will manually test the few endpoints to make sure all is ok.
Major worry would be the Date formatting.